src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -855,7 +868,21 @@ def _find_using_common_tenant(self, access_token, resource):
             subscriptions = self._find_using_specific_tenant(
                 tenant_id,
                 temp_credentials[_ACCESS_TOKEN])
-            all_subscriptions.extend(subscriptions)
+
+            # When a subscription can be listed by multiple tenants, only the first appearance is retained
+            for sub_to_add in subscriptions:
+                add_sub = True
+                for sub_to_compare in all_subscriptions:
+                    if sub_to_add.subscription_id == sub_to_compare.subscription_id:
+                        logger.warning("Subscription %s '%s' can be accessed from tenants %s(default) and %s. "
```

Comment:
For these new output formats, do we have some spec or context for this new multi-tenant behavior? I ask this because I want to learn if we should only output one subscription while it exists multi tenants or if we should output all of them and mark one of them as default one.

Reply:
We SHOULD and WILL display them all. But CLI is currently keying on `subscriptionId` in the account list which makes showing all of them impossible. This requires an overhaul of the account persistence mechanism.